lib/commands/location.js

```diff
@@ -1,27 +1,46 @@
+import _ from 'lodash';
 import { exec } from 'teen_process';
-import { fs, util } from 'appium-support';
+import { fs } from 'appium-support';
 import log from '../logger';
 
+const IDEVICELOCATION = 'idevicelocation';
+
 let commands = {};
 
+function formatLocationArg (value) {
+  value = `${value}`.trim();
+  // Negative coordinate values should be properly formatted
+  if (value.startsWith('-')) {
+    return ['--', value];
+  }
+  return [value];
+}
+
 commands.setGeoLocation = async function (location) {
   let {latitude, longitude} = location;
 
-  if (!util.hasValue(latitude) || !util.hasValue(longitude)) {
+  if (_.isEmpty(latitude) || _.isEmpty(longitude)) {
     log.errorAndThrow(`Both latitude and longitude should be set`);
   }
 
-  if (this.isRealDevice()) {
-    if (!(await fs.which('idevicelocation'))) {
-      log.errorAndThrow(`idevicelocation doesn't exist on the host`);
-    }
-    try {
-      await exec('idevicelocation', ['-u', this.opts.udid, `${latitude}`, `${longitude}`]);
-    } catch (e) {
-      throw new Error(`Can't set the location on device '${this.opts.udid}'. Original error: ${e.message}`);
-    }
-  } else {
+  if (this.isSimulator()) {
     await this.opts.device.setGeolocation(`${latitude}`, `${longitude}`);
+    return;
+  }
+
+  try {
+    await fs.which(IDEVICELOCATION);
+  } catch (e) {
+    log.errorAndThrow(`${IDEVICELOCATION} doesn\'t exist on the host. ` +
```

Comment:
No need to escape the apostrophe within the backtick quotes.

Reply:
fixed